src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -54,7 +55,23 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
     def refresh_repo(self):
         self.composite_logger.log("Refreshing local repo...")
         # self.invoke_package_manager(self.repo_clean)  # purges local metadata for rebuild - addresses a possible customer environment error
-        self.invoke_package_manager(self.repo_refresh)
+        for i in range(0, Constants.MAX_REPO_REFRESH_RETRY_COUNT):
+            try:
+                self.invoke_package_manager(self.repo_refresh)
+                return
+            except Exception as error:
+                if i < Constants.MAX_REPO_REFRESH_RETRY_COUNT - 1:
+                    self.composite_logger.log_warning("Exception on package manager refresh repo. [Exception={0}] [RetryCount={1}]".format(repr(error), str(i)))
+                    time.sleep(i + 1)
+                else:
+                    error_msg = "Unable to refresh repo (retries exhausted). [{0}] [RetryCount={1}]".format(repr(error), str(i))
+                    self.composite_logger.log_warning(error_msg)
+                    self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
+
+                    # Reboot if possible and not already done (TODO: find a way to detect if a reboot has been done already)
+                    self.composite_logger.log_warning("Setting force_reboot flag to True.")
+                    self.force_reboot = True
```

Comment:
Unsure of how to check if we have already rebooted from a past flag set here.

I am thinking that something like this could work:
```py
if self.status_handler.get_installation_reboot_status() == Constants.RebootStatus.COMPLETED:
```
but am unsure if other factors could affect it, such as other things causing a reboot before this reboot is triggered.

Reply:
You can condition it on the proposed check.

Reply:
Thanks, done.